linode/data_source_linode_domain_record.go

```diff
@@ -0,0 +1,140 @@
+package linode
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/linode/linodego"
+)
+
+func dataSourceLinodeDomainRecord() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLinodeDomainRecordRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
```

Comment:
TypeInt - also affects https://github.com/terraform-providers/terraform-provider-linode/pull/85/files#diff-7afd8b7c70ffcf69664d792604948e34R84-R86

Reply:
For example, https://github.com/terraform-providers/terraform-provider-linode/blob/master/linode/data_source_linode_volume.go#L16-L20 (there are other `id` fields that are String, so it wasn't clear if this was intentional. In other resources, like `image`, the `id` is actually a string.

Reply:
Ah okay. I definitely felt like this made most sense as an `int` but was trying to be consistent with the [other file](https://github.com/terraform-providers/terraform-provider-linode/blob/master/linode/data_source_linode_domain.go#L18) I was referencing.

We can start to make some progress on correcting these ids.

Reply:
I went ahead and fixed this and [this](https://github.com/terraform-providers/terraform-provider-linode/pull/85#discussion_r380328916).